src/test/java/edu/harvard/iq/dataverse/api/FilesIT.java

```diff
@@ -2483,4 +2485,122 @@ public void testCollectionStorageQuotas() {
+{
+    "label": "foo.txt"
+}
+""";
+
+        Response updateMetadataResponse = UtilIT.updateFileMetadata(fileId.toString(), updateJsonString, apiToken);
+        updateMetadataResponse.prettyPrint();
+        assertEquals(OK.getStatusCode(), updateMetadataResponse.getStatusCode());
+
+        Response getFileCitationPostV1Draft = UtilIT.getFileCitation(fileId, DS_VERSION_DRAFT, apiToken);
+        getFileCitationPostV1Draft.prettyPrint();
+        getFileCitationPostV1Draft.then().assertThat()
+                .statusCode(OK.getStatusCode())
+                .body("data.message", equalTo("Finch, Fiona, " + currentYear + ", \"Darwin's Finches\", <a href=\"" + pidAsUrl + "\" target=\"_blank\">" + pidAsUrl + "</a>, Root, DRAFT VERSION; foo.txt [fileName]"));
+
+        Response getFileCitationV1OldFilename = UtilIT.getFileCitation(fileId, "1.0", apiToken);
+        getFileCitationV1OldFilename.prettyPrint();
+        getFileCitationV1OldFilename.then().assertThat()
+                .statusCode(OK.getStatusCode())
+                .body("data.message", equalTo("Finch, Fiona, " + currentYear + ", \"Darwin's Finches\", <a href=\"" + pidAsUrl + "\" target=\"_blank\">" + pidAsUrl + "</a>, Root, V1; file.txt [fileName]"));
+
+        UtilIT.publishDatasetViaNativeApi(datasetId, "major", apiToken)
+                .then().assertThat().statusCode(OK.getStatusCode());
+
+        Response deaccessionDataset = UtilIT.deaccessionDataset(datasetId, "1.0", "just because", "http://example.com", apiToken);
+        deaccessionDataset.prettyPrint();
+        deaccessionDataset.then().assertThat().statusCode(OK.getStatusCode());
+
+        Response getFileCitationV1PostDeaccessionAuthor = UtilIT.getFileCitation(fileId, "1.0", apiToken);
+        getFileCitationV1PostDeaccessionAuthor.prettyPrint();
+        getFileCitationV1PostDeaccessionAuthor.then().assertThat()
+                .statusCode(OK.getStatusCode())
+                .body("data.message", equalTo("Finch, Fiona, " + currentYear + ", \"Darwin's Finches\", <a href=\"" + pidAsUrl + "\" target=\"_blank\">" + pidAsUrl + "</a>, Root, V1, DEACCESSIONED VERSION; file.txt [fileName]"));
+
+        Response getFileCitationV1PostDeaccessionNoApiToken = UtilIT.getFileCitation(fileId, "1.0", null);
+        getFileCitationV1PostDeaccessionNoApiToken.prettyPrint();
+        getFileCitationV1PostDeaccessionNoApiToken.then().assertThat()
+                .statusCode(UNAUTHORIZED.getStatusCode())
+                .body("message", equalTo("Dataset version cannot be found or unauthorized."));
+
+        Response getFileCitationV1PostDeaccessionNoPermsUser = UtilIT.getFileCitation(fileId, "1.0", noPermsApiToken);
+        getFileCitationV1PostDeaccessionNoPermsUser.prettyPrint();
+        getFileCitationV1PostDeaccessionNoPermsUser.then().assertThat()
+                .statusCode(UNAUTHORIZED.getStatusCode())
+                .body("message", equalTo("Dataset version cannot be found or unauthorized."));
+
+    }
```

Comment:
I am not seeing any test to cover the resulting citation string when the file is given a persistent ID.

I think we should test such case, as it is a requirement to support the current citation string formats in the SPA, as described in https://github.com/IQSS/dataverse-frontend/pull/269#issuecomment-1874193736.

Reply:
I agree but isn't it blocked by this issue?

- #6646

If Permalinks support file PIDs (I'm not sure), we should switch to that some day, and remove the FAKE DOI provider from the system.

Reply:
If we cannot test it then it is fine by me as long as the new endpoint still supports the way the citation text is composed for files with PIDs.

However, I think @MellyGray needed to set persistent IDs to files recently for testing, I'm not sure what mechanism she used.

We can talk about this at today's frontend weekly meeting. Maybe other folks can share their thoughts on this.

Reply:
I created some test files with persistentIds by using some API endpoints to register the test files, as [Leonid suggested](https://agilemonkeys.slack.com/archives/C010LA04BCG/p1706195174646599?thread_ts=1706193514.278269&cid=C010LA04BCG)

https://github.com/IQSS/dataverse-client-javascript/blob/d958605e6ea70f25d3283e7bc41fae3ec279b444/test/testHelpers/files/filesHelper.ts#L95

```javascript
export const registerFileViaApi = async (fileId: number): Promise<AxiosResponse> => {
  return await  enableFilePIDs().then(() => axios.get(`${TestConstants.TEST_API_URL}/admin/${fileId}/registerDataFile`, {
      headers: {
        'X-Dataverse-Key': process.env.TEST_API_KEY,
      }
  }));
}

const enableFilePIDs = async (): Promise<AxiosResponse> => {
  return await axios.put(`${TestConstants.TEST_API_URL}/admin/settings/:AllowEnablingFilePIDsPerCollection`, "true", {
    headers: {
      'X-Dataverse-Key': process.env.TEST_API_KEY,
    },
  }).then(() => axios.put(`${TestConstants.TEST_API_URL}/dataverses/root/attribute/filePIDsEnabled?value=true`, {}, {
    headers: {
      'X-Dataverse-Key': process.env.TEST_API_KEY,
    },
  }));
};
```

Reply:
I started poking at this and I'm glad you asked. In JSF when the file [has a PID](https://dataverse.harvard.edu/file.xhtml?persistentId=doi:10.7910/DVN/TJCLKP/3VSTKY) it is shown in the citation:

![Screenshot 2024-02-07 at 4 59 39 PM](https://github.com/IQSS/dataverse/assets/21006/ddb23331-8f47-4d3c-ba11-8913825a5a99)

Here's and example of [no file PID](https://dataverse.harvard.edu/file.xhtml?fileId=6867331&version=4.0):

![Screenshot 2024-02-07 at 4 59 46 PM](https://github.com/IQSS/dataverse/assets/21006/500112e8-3cab-4518-963e-62cae55fcaa4)

And the file citation is fairly different!

I'm in the middle of writing a test with FilePIDsEnabled but I'm not seeing the file PID so I need to go back and look at the code.

Reply:
@MellyGray hmm, I still can't create files with PIDs. You're using the FAKE DOI provider, right? (Again I thought it wasn't supported: #6646.)

Also, the Slack link above doesn't work for me. Do you think you could find the IQSS version?

@qqmyers when I run your `mvn test -Dtest=FilesIT#testFilePIDsBehavior` test, I don't get file PIDs either. `persistentId` is empty string.

If I switch to the DataCite DOI provider, I should be able to use test credentials to create file PIDs. However, I won't be able to add a test to the code.

I could also maybe try the Permanlink provider. I'm not sure if file PIDs are supported or not.

Reply:
FWIW - I think Fake will not create file DOIs still since it's createIdentifier method isn't called and it's publicize method just returns true. PermaLinks should work since publicize actually generates the PID if needed - https://github.com/IQSS/dataverse/blob/57e83168c213f1ecd9ecc769dd7596c890ee86fb/src/main/java/edu/harvard/iq/dataverse/pidproviders/PermaLinkPidProviderServiceBean.java#L110-L113 . (My old PR#7334 does register file PIDs as part of publication which I think would make Fake work for files - maybe that could be cherry-picked since the concern with the overall PR was slowing upload.)

Reply:
I couldn't get Permalinks working (Slack thread) but I was able to give a file a PID with DataCite. Unfortunately, my code isn't generating the right citation:

Expected (JSF):

<img width="770" alt="Screenshot 2024-02-08 at 4 38 44 PM" src="https://github.com/IQSS/dataverse/assets/21006/a87d64ca-de22-419d-be65-44d0ebac91a5">

Actual (this PR):

`curl -s "http://localhost:8080/api/files/14/versions/:latest-published/citation" | jq .data.message`

`"Admin, Dataverse, 2024, \"zzz\", <a href=\"https://doi.org/10.70122/FK2/XLJSYD\" target=\"_blank\">https://doi.org/10.70122/FK2/XLJSYD</a>, Root, V2; homer-dataverse.jpg [fileName]"`

I'll keep banging on it.

Reply:
Phew! I switch to test DataCite credentials, which let me set up files with PIDs. That allowed me to fix the code in 244cb1a7a3 and add some tests.